pennylane/devices/default_pulse.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+The default.pulse device builds on PennyLane's standard qubit-based device, adding support for
+pulse-based circuits.
+"""
+import numpy as np
+
+from .._version import __version__
+from .default_qubit import DefaultQubit
+
+
+class DefaultPulse(DefaultQubit):
+    """Default pulses device for PennyLane.
+
+    Args:
+        wires (int, Iterable[Number, str]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
+            or strings (``['ancilla', 'q1', 'q2']``). Default 1 if not specified.
+        shots (None, int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to ``None`` if not specified, which means that the device
+            returns analytical results.
+        dt (float): the time step used by the differential equation solver to evolve the
+             time-dependent Hamiltonian.
+        dim (int): dimensions of the system (2 for qubits, 3 for qutrits, etc)
+        drift(Operator): optional Hamiltonian term representing a constant drift term for the system Hamiltonian
```

Comment:
Should we perhaps omit these terms for now, until we're ready to use them?

Reply:
It would be also nice to raise a warning when drift is not None, saying that this currently does nothing